### PR TITLE
OSAC-261-Update Resource for FTP activities to match FIGMA designs

### PR DESCRIPTION
**Pull request by viseanadrianiulian**
Description
Update Resource for FTP activities to match FIGMA designs
How to test
Open Activity package in Studio Web and Studio Desktop
Jira
OSAC-261
Label:
24S

### CODE REVIEW

File: Activities/FTP/UiPath.FTP.Activities.Design/ActivitySynonymApiRegistration.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using UiPath.Studio.Activities.Api;
+using UiPath.FTP.Activities.Design.Properties;
+
+namespace UiPath.FTP.Activities.Design
+{
+    internal class ActivitySynonymApiRegistration : ApiRegistrationBase
```
Comment (alcxs) [inline Activities/FTP/UiPath.FTP.Activities.Design/ActivitySynonymApiRegistration.cs:10]: XML Comment on all public / internal members

File: Activities/FTP/UiPath.FTP.Activities.Design/DesignerMetadata.cs [hunk 2]
```diff
@@ -13,6 +14,26 @@ public class DesignerMetadata : IRegisterMetadata
...
+        {
+            if (api is IWorkflowDesignApi wfDesignApi)
+            {
+                _wfDesignApi = wfDesignApi;
+
+                new ActivitySynonymApiRegistration().Initialize(wfDesignApi);
+            }
+        }
+
+        public void Initialize(object api)
```
Comment (alcxs) [inline Activities/FTP/UiPath.FTP.Activities.Design/DesignerMetadata.cs:34]: XML Comment

File: Activities/FTP/UiPath.FTP.Activities.Design/DesignerMetadata.cs [hunk 3]
```diff
@@ -13,6 +14,26 @@ public class DesignerMetadata : IRegisterMetadata
...
+        {
+            if (api == null)
+            {
+                return;
+            }
+            InitializeInternal(api);
+        }
         public void Register()
```
Comment (alcxs) [inline Activities/FTP/UiPath.FTP.Activities.Design/DesignerMetadata.cs:43]: add a space before this line

File: Activities/FTP/UiPath.FTP.Activities.Design/ApiRegistrationBase.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using UiPath.Studio.Activities.Api;
+
+namespace UiPath.FTP.Activities.Design
+{
+    internal abstract class ApiRegistrationBase
```
Comment (alcxs) [inline Activities/FTP/UiPath.FTP.Activities.Design/ApiRegistrationBase.cs:10]: XML Comment on all public / internal members

**Review by alcxs — CHANGES_REQUESTED** [review]
minor styling changes and comments

File: Activities/FTP/UiPath.FTP.Activities.Design/ActivitySynonymApiRegistration.cs (hunk 1 above)
Comment (viseanadrianiulian) [reply Activities/FTP/UiPath.FTP.Activities.Design/ActivitySynonymApiRegistration.cs:10]: done

File: Activities/FTP/UiPath.FTP.Activities.Design/DesignerMetadata.cs (hunk 3 above)
Comment (viseanadrianiulian) [reply Activities/FTP/UiPath.FTP.Activities.Design/DesignerMetadata.cs:43]: done

File: Activities/FTP/UiPath.FTP.Activities.Design/ApiRegistrationBase.cs (hunk 4 above)
Comment (viseanadrianiulian) [reply Activities/FTP/UiPath.FTP.Activities.Design/ApiRegistrationBase.cs:10]: done

File: Activities/FTP/UiPath.FTP.Activities.Design/DesignerMetadata.cs (hunk 2 above)
Comment (viseanadrianiulian) [reply Activities/FTP/UiPath.FTP.Activities.Design/DesignerMetadata.cs:34]: done

**Review by alcxs — APPROVED** [review]

File: Activities/FTP/UiPath.FTP.Activities.Design/ActivitySynonymApiRegistration.cs [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        public override bool CanPerformRegistration(IWorkflowDesignApi api)
+        {
+            return api != null && api.HasFeature(DesignFeatureKeys.ActivitySynonyms);
+        }
+
+        /// <summary>
+        /// Register synonyms for the required activities
+        /// </summary>
+        /// <param name="api">The api</param>
+        protected override void PerformRegistration(IWorkflowDesignApi api)
```
Comment (ovidiuiliescu) [inline Activities/FTP/UiPath.FTP.Activities.Design/ActivitySynonymApiRegistration.cs:27]: Please add `[MethodImpl(MethodImplOptions.NoInlining)]` to this method.

See how other activity packs do this (e.g. Mail in the main Activities repo).

**Review by ovidiuiliescu — CHANGES_REQUESTED** [review]

**Review by ovidiuiliescu — APPROVED** [review]

**Pull request merged**